src/components/Main.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useContext } from 'react';
+import { TodoList } from './TodoList';
+import { TodosContext } from '../contexts/TodosContext';
+import { Todo } from '../types/Todo';
+
+export const Main: React.FC = () => {
+  const { todos, setTodos } = useContext(TodosContext);
+
+  const isChecked = todos.every(todo => todo.completed);
+
+  const handleToggleAll = () => {
+    let modifiedTodos: Todo[] = [];
+    const areAllSameTodoStatus
+      = todos.every(todo => todo.completed)
+      || todos.every(todo => !todo.completed);
+
+    if (areAllSameTodoStatus) {
+      modifiedTodos = todos.map(todo => ({
+        ...todo,
+        completed: !todo.completed,
+      }));
+    } else {
+      modifiedTodos = todos.map(todo => ({
+        ...todo,
+        completed: true,
+      }));
+    }
```

Comment:
```suggestion
  const isAllCompleted = todos.every(todo => todo.completed);
  const modifiedTodos = todos.map(todo => ({
    ...todo,
    completed: !isAllCompleted,
  }));
```

should work the same but simplify, check please